src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1780,6 +1785,50 @@ def _encode_custom_key_description(key_description):
     return key_description.encode('utf-16')
 
 
+def _resolve_assignee_object(cli_ctx, assignee, assignee_object_id, assignee_principal_type):
+    client = _graph_client_factory(cli_ctx)
+    result = None
+
+    # resolve user name/service principal assignee
+    if assignee:
+        if assignee.find('@') >= 0:  # looks like a user principal name
+            result = list(client.users.list(filter="userPrincipalName eq '{}'".format(assignee)))
+        if not result:
+            result = list(client.service_principals.list(
+                filter="servicePrincipalNames/any(c:c eq '{}')".format(assignee)))
+        if result:
+            return result[0].object_id, result[0].object_type
+
+    # verify object id
+    try:
+        if assignee_object_id:
+            result = _get_object_stubs(client, [assignee_object_id])
+        elif is_guid(assignee):
+            result = _get_object_stubs(client, [assignee])
```

Comment:
`fallback_to_object_id` by default is `False`. 

https://github.com/Azure/azure-cli/blob/5544a0f3bc32aa778d296f71c4d37e48f8ebe691/src/azure-cli/azure/cli/command_modules/role/custom.py#L1783

Using `assignee` as object ID causes a breaking change.